tests/programs/dlopen_demo.sh

```diff
@@ -6,17 +6,29 @@
 # Copyright The Mbed TLS Contributors
 # SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
 
-. "${0%/*}/../../scripts/demo_common.sh"
+SCRIPT_DIR=$(dirname "$0")
+. "${SCRIPT_DIR}/../../scripts/demo_common.sh"
```

Comment:
In a pure TF-PSA-Crypto context (not TF-PSA-Crypto inside Mbed TLS), when I run `./tests/scripts/all.sh test_tf_psa_crypto_cmake_shared` with https://github.com/Mbed-TLS/TF-PSA-Crypto/pull/181, I get `This doesn't seem to be an Mbed TLS source tree.` when running `$FRAMEWORK/tests/programs/dlopen_demo.sh`. I think this is because `demo_common.sh` needs to be adapted to TF-PSA-Crypto.